src/pointcloud_to_laserscan_node.cpp

```diff
@@ -225,9 +243,80 @@ void PointCloudToLaserScanNode::cloudCallback(
     int index = (angle - scan_msg->angle_min) / scan_msg->angle_increment;
     if (range < scan_msg->ranges[index]) {
       scan_msg->ranges[index] = range;
+      v_pointcloud_index.at(index) = pointcloud_index;
     }
   }
-  pub_->publish(std::move(scan_msg));
+
+  laserscan_pub_->publish(std::move(scan_msg));
+
+  pcl::PointCloud<pcl::PointXYZ> pcl_pointcloud;
+  sensor_msgs::PointCloud2ConstIterator<float> iter_x(*cloud_msg, "x"), iter_y(*cloud_msg, "y"),
+    iter_z(*cloud_msg, "z");
+  for (size_t i = 0; i < v_pointcloud_index.size(); ++i) {
+    if (v_pointcloud_index.at(i) != no_data) {
+      pcl::PointXYZ point;
+      point.x = *(iter_x + v_pointcloud_index.at(i));
+      point.y = *(iter_y + v_pointcloud_index.at(i));
+      point.z = *(iter_z + v_pointcloud_index.at(i));
+      pcl_pointcloud.push_back(point);
+    }
+  }
+  pcl::toROSMsg(pcl_pointcloud, pointcloud_output);
+  auto pointcloud_output_ptr
```

Comment:
[nits] const?

Reply:
Const unique pointer is prohibited from being moved?
```
error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = sensor_msgs::msg::PointCloud2_<std::allocator<void> >; _Dp = std::default_delete<sensor_msgs::msg::PointCloud2_<std::allocator<void> > >]’
  270 |   pointcloud_pub_->publish(std::move(pointcloud_output_ptr));
```

Reply:
I'm sorry, okay let's leave it as it is!